### DISCUSSION TITLE
EAS, Expo

### DISCUSSION THREAD

**Discussion by Takapy1** [Ayato-kosaka/react-itinerary#877] [General]
EAS, Expo関連のナレッジを貯めていきます。

**Comment by Takapy1** [comment]
# EASの無料枠について
## EAS Buildの無料枠
以下のように30回のBuildが可能だが、そのうち**iOSは最大15回まで**となっている。

> Builds: Get started with 30 lower-priority builds (of those, up to 15 iOS) per month
> Concurrencies: 1
> Build timeout: 45 minutes
[【公式】 EAS Buildの無料枠](https://expo.dev/accounts/spelieve/settings/billing)

## 無料枠を超えた場合のエラー
Build出来なくなる

```bash
$ eas build --profile development --platform ios

We've encountered a transient error. Try again shortly.
Build request failed. Make sure you are using the latest eas-cli version. If the problem persists, report the issue.
    CombinedError: [GraphQL] This account has used its iOS builds from the Free plan this month, which will reset in 16 days (on Wed Nov 01 2023). Upgrade your plan for more builds with shorter wait times and longer timeouts, and to run 
    more builds concurrently with a subscription plan. https://expo.dev/accounts/spelieve/settings/billing
```


# 超えた場合の解決方法
## ローカルビルドをする（まだ実行できてない！！！）
[公式](https://docs.expo.dev/build-reference/local-builds/)
[わかりやすい記事](https://zenn.dev/tadaedo/scraps/6f989f2a104f76)
```
$ eas build --profile development --platform ios --local
```

**Reply by Ayato-kosaka** [reply]
@Takapy1 
まじか、、制限あるの知らなかった、、
Github Action のせいで流れまくるから、Submit 以外 push 起点で流すの止めるね。
https://github.com/Ayato-kosaka/spelieve/pull/878
で、PL作って修正した！
